src/main/java/at/ac/tuwien/kr/alpha/api/externals/stdlib/AspStandardLibrary.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) 2020, the Alpha Team.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1) Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2) Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package at.ac.tuwien.kr.alpha.api.externals.stdlib;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Set;
+
+import at.ac.tuwien.kr.alpha.api.externals.ExternalUtils;
+import at.ac.tuwien.kr.alpha.api.externals.Predicate;
+import at.ac.tuwien.kr.alpha.common.terms.ConstantTerm;
+import at.ac.tuwien.kr.alpha.common.terms.Terms;
+
+/**
+ * Collection of methods that can be used as external atoms from ASP programs.
+ * Provides commonly used functionality such as basic string operations,
+ * datetime handling etc.
+ * 
+ * All functions exposed by this class are guaranteed to be stateless and
+ * side-effect free.
+ * 
+ * Copyright (c) 2020, the Alpha Team.
+ */
+public final class AspStandardLibrary {
+
+	private AspStandardLibrary() {
+
+	}
+
```

Comment:
We should consider adding standard list operations (construct/head/tail/member/append/length) also to the library.

Reply:
I'd be a fan of that. However, I think construct and append shouldn't be part of (at least the first iteration of) list support. Both operations could be understood as "write operations", therefore adding a procedural dimension to programs that's hard to handle with current ASP semantics.
Should we use Issue #76 to track ideas about that? Seems like the right place..

Reply:
Well, without construct I don't see a way to construct a list ... ?
By the way, in Prolog a list is just a special function term, so `[a,b,c]` is equal to `cons(a, cons(b, cons(c, nil)))`. However, if we implement lists via external atoms we have to be careful whenever lists are modified, because logically the modified list is a new object, i.e., in such cases the underlying Java list must be duplicated too.

Reply:
The problem I see with construction of lists via externals is this:
```
% forgive the syntax, just prototyping here...
a_list_as_fact(lst[1, 2, 3]). % no problem, list is a fact and will always exist

% Assume that LST is a (list-)function term, the rule reads content from SRC using 
% the external atom ext_content_list when some_other_condition(SRC) holds.
content_list(LST) :- content_src(SRC), &ext_content_list[SRC](LST), some_other_condition(SRC).
% This is a problem! Since the external atom is evaluated when the rule is grounded
% - but before it is clear whether it can actually fire - every failed attempt to ground this rule
% will create a new (Java-)list instance on the heap. 
```
The useless list instances mentioned should (I think, without having gone through the code with a fine-toothed comb..) be garbage-collected at some point, but I still see a potential memory leak here, or at least a dangerous (even if  temporary) heap-bloat.

Reply:
WIthout completely following this example, I still agree that native lists might come with intricate memory issues. Maybe we should discuss any further thoughts on that on #76 as you suggested.